lib/instance/bloc/instance_bloc.dart

```diff
@@ -0,0 +1,66 @@
+import 'package:bloc/bloc.dart';
+import 'package:bloc_concurrency/bloc_concurrency.dart';
+import 'package:equatable/equatable.dart';
+import 'package:lemmy_api_client/v3.dart';
+import 'package:stream_transform/stream_transform.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+
+import 'package:thunder/instance/enums/instance_action.dart';
+import 'package:thunder/core/singletons/lemmy_client.dart';
+import 'package:thunder/instance/utils/instance.dart';
+import 'package:thunder/utils/global_context.dart';
+
+part 'instance_event.dart';
+part 'instance_state.dart';
+
+const throttleDuration = Duration(milliseconds: 100);
+
+EventTransformer<E> throttleDroppable<E>(Duration duration) {
+  return (events, mapper) {
+    return droppable<E>().call(events.throttle(duration), mapper);
+  };
+}
+
+class InstanceBloc extends Bloc<InstanceEvent, InstanceState> {
+  final LemmyClient lemmyClient;
+
+  InstanceBloc({required this.lemmyClient}) : super(const InstanceState()) {
+    /// Handles clearing any messages from the state
+    on<InstanceClearMessageEvent>(
+      _onInstanceClearMessage,
+      transformer: throttleDroppable(Duration.zero),
+    );
+
+    /// Handles actions related to an instance
+    on<InstanceActionEvent>(
+      _onInstanceAction,
+      transformer: throttleDroppable(Duration.zero),
+    );
+  }
+
+  /// Handles clearing any messages from the state
+  Future<void> _onInstanceClearMessage(InstanceClearMessageEvent event, Emitter<InstanceState> emit) async {
+    emit(state.copyWith(status: InstanceStatus.success, message: null));
+  }
+
+  /// Handles instance related actions
+  Future<void> _onInstanceAction(InstanceActionEvent event, Emitter<InstanceState> emit) async {
+    emit(state.copyWith(status: InstanceStatus.fetching));
+
+    final l10n = AppLocalizations.of(GlobalContext.context)!;
+
+    switch (event.instanceAction) {
```

Comment:
I like this pattern of having one event and an enum of possible actions! Seems like it could simplify things.

Reply:
This pattern was also used in the feed refactor so I just used the same pattern here!